src/cli/commands/add.js

```diff
@@ -69,7 +69,11 @@ export class Add extends Install {
    * returns version for a pattern based on Manifest
    */
   getPatternVersion(pattern: string, pkg: Manifest): string {
-    const {exact, tilde} = this.flags;
+    const tilde = this.flags.tilde;
+    const exact =
+      this.flags.exact ||
+      Boolean(this.config.getOption('save-exact')) ||
+      String(this.config.getOption('save-prefix')) === '';
```

Comment:
Assigning this to a variable like `configPrefix` might be good since it is now used here and also below.